### PR TITLE
📖 Improve provider definition in the glossary

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:

This PR updates the definition of provider in the glossary to include all the components that can be deployed and provides a functionality to the Cluster API management Cluster.

This acknowledges the fact that we are already using this term for many things including core provider, infra provider, bootstrap and CP providers, and makes this suitable for including IPAM providers, Cluster add-on providers, Runtime Extension providers (which is a new term being defined in the Runtime SDK proposal).

If this is accepted, this will allow us to benefit of all the clusterctl machinery and provide a consistent UX for the users

### CODE REVIEW

**Comment by fabriziopandini** [comment]
/hold 
for office hours discussion

**Comment by fabriziopandini** [comment]
/hold cancel
from the office hours discussion It seems there is a general consensus about the idea
I'm going to check the book to make sure we are using updated terms consistently, and finally remove wip

**Comment by fabriziopandini** [comment]
made a pass on the book ensuring terms are used consistently
now this PR is ready for review

File: docs/book/src/reference/glossary.md
Comment (sbueringer) [inline docs/book/src/reference/glossary.md:334]:
```suggestion
Runtime Extensions providers interaction with Cluster API are based on the Open API spec for [runtime hooks](#runtime-hook).
```

Comment (sbueringer) [reply docs/book/src/reference/glossary.md:334]: Or maybe Runtime Extension providers (?) (if yes, same in header)

**Review by sbueringer — COMMENTED** [review]
Just nits, otherwise lgtm

**Comment by sbueringer** [comment]
lgtm pending squash

File: docs/book/src/reference/glossary.md
Comment (killianmuldoon) [inline docs/book/src/reference/glossary.md:29]:
```suggestion
Bootstrap provider's interaction with Cluster API is based on what is defined in the [Cluster API contract](#contract). 
```

File: docs/book/src/reference/glossary.md
Comment (killianmuldoon) [inline docs/book/src/reference/glossary.md:138]:
```suggestion
Control plane provider's interaction with Cluster API is based on what is defined in the [Cluster API contract](#contract).
```

File: docs/book/src/reference/glossary.md
Comment (killianmuldoon) [inline docs/book/src/reference/glossary.md:157]:
```suggestion
A feature implementation offered as part of the Cluster API project and maintained by the CAPI core team. For example [KCP](#KCP) is a default implementation for
a [control plane provider](#control-plane-provider) .
```

File: docs/book/src/reference/glossary.md
Comment (killianmuldoon) [inline docs/book/src/reference/glossary.md:193]:
```suggestion
Clouds infrastructure providers include AWS, Azure, Google while VMware, MAAS and metal3.io are bare metal providers. When there is more than one way to obtain resources from the same infrastructure provider (e.g. EC2 vs. EKS in AWS) each way is referred to as a variant.
```

File: docs/book/src/reference/glossary.md
Comment (killianmuldoon) [inline docs/book/src/reference/glossary.md:190]:
```suggestion
Infrastructure provider's interaction with Cluster API is based on what is defined in the [Cluster API contract](#contract).
```

File: docs/book/src/reference/glossary.md
Comment (killianmuldoon) [inline docs/book/src/reference/glossary.md:197]:
```suggestion
See e.g. [CAPA][#CAPA], [CAPC][#CAPC], [CAPD][#CAPD], [CAPG][#CAPG], [CAPH][#CAPH], [CAPIBM][#CAPIBM], [CAPN][#CAPN], 
[CAPX][#CAPX], [CAPK][#CAPK], [CAPO][#CAPO], [CAPOCI][#CAPOCI], [CAPV][#CAPV], [CAPVC][#CAPVC], [CAPVCD][#CAPVCD], [CAPZ][#CAPZ]
```

File: docs/book/src/reference/glossary.md
Comment (killianmuldoon) [inline docs/book/src/reference/glossary.md:214]:
```suggestion
IPAM provider's interaction with Cluster API is based on the `IPAddressClaim` and `IPAddress` API types.
```

File: docs/book/src/reference/glossary.md
Comment (killianmuldoon) [inline docs/book/src/reference/glossary.md:303]:
```suggestion
to refer to any project that can be deployed and provides functionality to the Cluster API management Cluster.
```

File: docs/book/src/reference/glossary.md
Comment (killianmuldoon) [inline docs/book/src/reference/glossary.md:334]:
```suggestion
Runtime Extension provider's interaction with Cluster API are based on the Open API spec for [runtime hooks](#runtime-hook).
```

**Review by killianmuldoon — COMMENTED** [review]
Good changes IMO.

Just a bunch of nits and rewordings in my review, none of them blocking.

**Comment by sbueringer** [comment]
Thx!
/lgtm

File: docs/book/src/reference/glossary.md
Comment (vincepri) [inline docs/book/src/reference/glossary.md:97]:
```suggestion
Refers to an information technology (IT) company that provides computing resources (e.g. AWS, Azure, Google, etc.)
```

File: docs/book/src/reference/glossary.md
Comment (vincepri) [inline docs/book/src/reference/glossary.md:126]:
```suggestion
Defines a set of rules a [provider](#provider) is expected to comply with in order to interact with Cluster API.
Those rules can be in the form of CustomResourceDefinition (CRD) fields and/or expected behaviors to be implemented.
```

File: docs/book/src/reference/glossary.md
Comment (vincepri) [inline docs/book/src/reference/glossary.md:192]:
```suggestion
Clouds infrastructure providers include AWS, Azure, or Google; while VMware, MAAS, or metal3.io can be defined as bare metal providers. 
When there is more than one way to obtain resources from the same infrastructure provider (e.g. EC2 vs. EKS in AWS) each way is referred to as a variant.
```

**Review by killianmuldoon — APPROVED** [review]
/lgtm

**Comment by sbueringer** [comment]
@fabriziopandini Should we merge or do we want to wait a bit longer?

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**
